lib/polyfill/patchedmediakeys_20140218.js

```diff
@@ -0,0 +1,800 @@
+        if (cap.contentType) {
+          ranAnyTests = true;
+          var contentType = cap.contentType.split(';')[0];
+          if (MSMediaKeys.isTypeSupported(this.keySystem, contentType)) {
+            newCfg.audioCapabilities.push(cap);
+            success = true;
+          }
+        }
+      }
+    }
+    if (cfg.videoCapabilities) {
+      for (var j = 0; j < cfg.videoCapabilities.length; ++j) {
+        var cap = cfg.videoCapabilities[j];
+        if (cap.contentType) {
+          ranAnyTests = true;
+          var contentType = cap.contentType.split(';')[0];
+          if (MSMediaKeys.isTypeSupported(this.keySystem, contentType)) {
+            newCfg.videoCapabilities.push(cap);
+            success = true;
+          }
+        }
+      }
+    }
+
+    if (!ranAnyTests) {
+      // If no specific types were requested, we check all common types to find
+      // out if the key system is present at all.
+      success = MSMediaKeys.isTypeSupported(this.keySystem, 'video/mp4');
+    }
+    if (cfg.persistentState == 'required') {
+      if (allowPersistentState) {
+        newCfg.persistentState = 'required';
+      } else {
+        success = false;
+      }
+    }
+
+    if (success) {
+      this.configuration_ = newCfg;
+      return;
+    }
+  }  // for each cfg in supportedConfigurations
+
+  // As per the spec, this should be a DOMException, but
+  // there is not a public constructor for this
+  var unsupportedKeySystemError = new Error('Unsupported keySystem');
+  unsupportedKeySystemError.name = 'NotSupportedError';
+  unsupportedKeySystemError.code = DOMException.NOT_SUPPORTED_ERR;
+  throw unsupportedKeySystemError;
+};
+
+
+/** @override */
+shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeySystemAccess.prototype.
+    createMediaKeys = function() {
+  shaka.log.debug('v20140218.MediaKeySystemAccess.createMediaKeys');
+
+  // Alias
+  var v20140218 = shaka.polyfill.PatchedMediaKeys.v20140218;
+
+  var mediaKeys = new v20140218.MediaKeys(this.keySystem);
+  return Promise.resolve(/** @type {!MediaKeys} */ (mediaKeys));
+};
+
+
+/** @override */
+shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeySystemAccess.prototype.
+    getConfiguration = function() {
+  shaka.log.debug('v20140218.MediaKeySystemAccess.getConfiguration');
+  return this.configuration_;
+};
+
+
+/**
+ * @this {!HTMLMediaElement}
+ * @override
+ * */
+shaka.polyfill.PatchedMediaKeys.v20140218.setMediaKeys = function(mediaKeys_) {
+  shaka.log.debug('v20140218.setMediaKeys');
+
+  // If mediaKeys is null, ensure mediaKeys on the videoElement is null
+  if (!mediaKeys_) {
+    delete this['mediaKeys'];
+    this['mediaKeys'] = null;
+
+    return;
+  }
+
+  // If mediaKeys equals videoElement's mediaKey instance, do nothing
+  if (this.mediaKeys == mediaKeys_) {
+    return;
+  }
+
+  // Alias
+  var v20140218 = shaka.polyfill.PatchedMediaKeys.v20140218;
+
+  // Remove any old listeners.
+  mediaKeys_.eventManager_.removeAll();
```

Comment:
This is accessing a private variable on mediaKeys.  The compiler must not be enforcing these restriction correctly, but I can't quite figure out why.

In the v0.1b polyfill, we have a type cast and a public method on our MediaKeys implementation called setMedia to handle this without accessing private members.